app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java

```diff
@@ -469,13 +476,19 @@ public boolean test(@io.reactivex.annotations.NonNull String s) throws Exception
                 boolean lengthCheck = s.length() >= THRESHOLD_SUGGESTION;
                 // Clear the suggestions adapter if the length check fails
                 if (!lengthCheck && !suggestionListAdapter.isEmpty()) {
-                    suggestionListAdapter.updateAdapter(new ArrayList<String>());
+                    suggestionListAdapter.clearAdapter();
                 }
                 // Only pass through if suggestions is enabled and the query length is equal or greater than THRESHOLD_SUGGESTION
                 return showSuggestions && lengthCheck;
             }
         };
 
+        // TODO: this isn't working correctly
+        if (searchQuery.length() < THRESHOLD_SUGGESTION) {
+            List<SearchHistoryEntry> historyItems = searchHistoryDAO.getItemsForQuery(searchQuery + "%", 10).blockingFirst();
```

Comment:
Don't use `blockingFirst`, instead try to integrate this with the RxJava statements below.